src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -374,19 +400,19 @@ def get_upload_rate_limit(self, hops=None):
         # Rate conversion due to the fact that we had a different system with Swift
         # and the old python BitTorrent core: unlimited == 0, stop == -1, else rate in kbytes
         libtorrent_rate = self.get_session(hops).upload_rate_limit()
-        return 0 if libtorrent_rate == -1 else (-1 if libtorrent_rate == 1 else libtorrent_rate / 1024)
+        return self.__class__.reverse_convert_rate(rate=libtorrent_rate)
 
     def set_download_rate_limit(self, rate, hops=None):
-        libtorrent_rate = int(-1 if rate == 0 else (1 if rate == -1 else rate * 1024))
+        libtorrent_rate = self.__class__.convert_rate(rate=rate)
 
         # Pass outgoing_port and num_outgoing_ports to dict due to bug in libtorrent 0.16.18
         settings_dict = {'download_rate_limit': libtorrent_rate}
         for session in self.ltsessions.values():
             self.set_session_settings(session, settings_dict)
 
     def get_download_rate_limit(self, hops=0):
-        libtorrent_rate = self.get_session(hops).download_rate_limit()
-        return 0 if libtorrent_rate == -1 else (-1 if libtorrent_rate == 1 else libtorrent_rate / 1024)
+        libtorrent_rate = self.get_session(hops=hops).download_rate_limit()
+        return self.__class__.reverse_convert_rate(rate=libtorrent_rate)
```

Comment:
This version is a bit easier to read:

```suggestion
        return self.reverse_convert_rate(rate=libtorrent_rate)
```